src/merkle/store/mod.rs

```diff
@@ -257,6 +257,15 @@ impl MerkleStore {
         Ok(tree_depth)
     }
 
+    /// Iterator over the inner nodes of the [MerkleStore].
+    pub fn inner_nodes(&self) -> impl Iterator<Item = InnerNodeInfo> + '_ {
+        self.nodes.values().map(|&e| InnerNodeInfo {
+            value: Rpo256::merge(&[e.left, e.right]).into(),
+            left: e.left.into(),
+            right: e.right.into(),
+        })
```

Comment:
I would do this as:
```Rust
self.nodes.iter().map(|(r, &n)| InnerNodeInfo {
    value: r.into(),
    left: n.left.into(),
    right: n.right.into(),
})
```
This would avoid the need for hashing.